tools/performance-benchmarking/src/md/0010-content.md

```diff
@@ -0,0 +1,916 @@
+* run the tool with sensible defaults, or hook into every stage of the pipeline when you need more flexibility.
+
+This documentation contains three sections.
+The [user walkthrough](#user-walkthrough) takes you through the process of setting up an entirely new benchmark run and dashboard.
+The [developer reference](#developer-reference) describes `benchcomp`'s architecture and the different data formats that it uses, enabling you to author a benchmark run and dashboard of your own.
+The [visualization viewer guide](#benchcomp-visualization-viewer-guide) describes how to use dashboards that `benchcomp` generates.
+Although dashboards are highly customizable, the guide describes the common elements.
+
+
+# User Walkthrough
+
+This section presents a single user story, showing how the user can quickly get started with a simple benchmark comparison run, through to a more complex setup with extensive customization.
+This section explains the syntax in prose without attempting to exhaustively document it; the [Development Reference](#developer-reference) below is a full reference for the syntax.
+
+
+## Comparing two variants
+
+Alice wants to compare the performance of two versions of CBMC: the latest release, with and without a new optimization that she's implemented.
+Initially, Alice tries to do this entirely using `benchcomp`'s built-in parsers and filters.
+She performs the following steps:
+
+* Create a directory containing all AWS CBMC proofs, with a top-level script that runs all of them called `run-all-cbmc-proofs.py`
+* Create a configuration file (shown below)
+* Run `benchcomp -o result.json`
+
+The initial configuration file is:
+
+<div class="twocolumn">
+<div class="col col-66">
+
+```yaml
+# benchcomp.yaml
+
+run:
+  suites:
+    all_cbmc_proofs:
+      parser:
+        type: built_in
+        name: litani_to_benchcomp
+        directory: all_cbmc_proofs
+      variants:
+        optimized:
+          provenance: inline
+          config:
+            command_line: ./run-cbmc-proofs.py
+            env:
+              CBMC: ~/src/cbmc/build/bin/cbmc
+        release:
+          provenance: inline
+          config:
+            command_line: ./run-cbmc-proofs.py
+            env:
+              CBMC: /usr/local/bin/cbmc
+```
+
+</div>
+<div class="col col-33">
+
+This says: run two *variants* (`optimized` and `release`) of a single benchmark *suite* (`all_cbmc_proofs`).
+The variants are distinguished in this case by the `CBMC` environment variable, which the AWS [proof build system](https://github.com/model-checking/cbmc-starter-kit) uses when invoking CBMC.
+
+</div>
+</div>
+
+`benchcomp` copies the `all_cbmc_proofs` directory to two temporary directories, one for each variant, and runs the command line. It uses the built-in `litani_to_benchcomp` parser to assemble the results. `benchcomp` then writes this data to the output file in JSON format (here in YAML for readability):
```

Comment:
Also, isn't the `JSON` an intermediate step for benchcomp at this point? Should we even talk about it in this use case?